File: SuperiorPackGroup/BLL/EmployeesBLL.vb
Imports DXDAL.SPGData
Imports DevExpress.Xpo

Public Class EmployeesBLL

    Public Shared Function DeleteEmployee(ByVal employeeID As Integer) As Boolean

        Dim employee As Employees = Session.DefaultSession.GetObjectByKey(Of Employees)(employeeID)

        If employee.CheckedReceiving.Count > 0 OrElse employee.CheckedShipping.Count > 0 OrElse employee.Loaded.Count > 0 OrElse employee.Unloaded.Count > 0 OrElse employee.CheckedTransfer.Count > 0 OrElse employee.LoadedTransfer.Count > 0 OrElse
                employee.UnloadedTransfer.Count > 0 Then
            MessageBox.Show("You can't delete this employee there are other records associated to it")
        Else
            employee.Delete()
        End If

        Return employee.IsDeleted

    End Function

End Class

File: SuperiorPackGroup/BLL/EmployeesBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using DXDAL.SPGData;
using DevExpress.Xpo;

namespace SuperiorPackGroup
{
	public class EmployeesBLL
	{

		public static bool DeleteEmployee(int employeeID)
		{

			Employees employee = Session.DefaultSession.GetObjectByKey<Employees>(employeeID);

			if (employee.CheckedReceiving.Count > 0 || employee.CheckedShipping.Count > 0 || employee.Loaded.Count > 0 || employee.Unloaded.Count > 0 || employee.CheckedTransfer.Count > 0 || employee.LoadedTransfer.Count > 0 || employee.UnloadedTransfer.Count > 0)
			{
				MessageBox.Show("You can't delete this employee there are other records associated to it");
			}
			else
			{
				employee.Delete();
			}

			return employee.IsDeleted;

		}

	}

}